src/System.Data.SqlClient/src/Interop/SNINativeMethodWrapper.cs

```diff
@@ -10,6 +10,7 @@ internal static class SNINativeMethodWrapper
     {
         private const string SNI = "sni.dll";
 
+#if !MANAGED_SNI
```

Comment:
If we have multiple files laying around, then we'll start encountering a lot of maintenance issues later on. The code would also have to be pulled apart in such a way that it becomes un unreadable spaghetti mess. Using #ifdef controls is a much more elegant solution for us, and is what we do in Azure code anyway.